File: Test_Coverage/processed_dataset/tests/test_HumanEval_124.py
from functions.HumanEval_124 import valid_date

def test_HumanEval_124():
  assert valid_date('06-04-2020') == True
  
  assert valid_date('01-01-2007') == True
  
  assert valid_date('03-32-2011') == False
  
  assert valid_date('') == False
  
  assert valid_date('04-31-3000') == False
  
  assert valid_date('06-06-2005') == True
  
  assert valid_date('21-31-2000') == False
  
  assert valid_date('04-12-2003') == True
  
  assert valid_date('04122003') == False
  
  assert valid_date('20030412') == False
  
  assert valid_date('2003-04') == False
  
  assert valid_date('2003-04-12') == False
  
  assert valid_date('04-2003') == False

File: Test_Coverage_Java/src/test/java/test_HumanEval_124.java
import org.junit.Test;
import static org.junit.Assert.*;

public class HumanEval_124Test {

    private HumanEval_124 humanEval = new HumanEval_124();

    @Test
    public void testValidDate_SimpleCases() {
        assertTrue(humanEval.validDate("03-11-2000"));
        assertFalse(humanEval.validDate("15-01-2012"));
        assertFalse(humanEval.validDate("04-0-2040"));
        assertTrue(humanEval.validDate("06-04-2020"));
        assertTrue(humanEval.validDate("01-01-2007"));
        assertFalse(humanEval.validDate("03-32-2011"));
        assertFalse(humanEval.validDate(""));
        assertFalse(humanEval.validDate("04-31-3000"));
        assertTrue(humanEval.validDate("06-06-2005"));
        assertFalse(humanEval.validDate("21-31-2000"));
        assertTrue(humanEval.validDate("04-12-2003"));
        assertFalse(humanEval.validDate("04122003"));
        assertFalse(humanEval.validDate("20030412"));
        assertFalse(humanEval.validDate("2003-04"));
        assertFalse(humanEval.validDate("2003-04-12"));
        assertFalse(humanEval.validDate("04-2003"));
    }

}